### PR TITLE
Adjust tests in test_node_replacement_proactive.py to run on vSphere IPI

### PR DESCRIPTION
**Pull request by sidhant-agrawal**
Fixes: #4190 

Signed-off-by: Sidhant Agrawal <sagrawal@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/node.py [hunk 1]
```diff
@@ -1481,9 +1475,12 @@ def wait_for_new_osd_node(old_osd_node_names, timeout=600):
             Else it returns None
 
     """
+    pod.wait_for_pods_to_be_running(
+        pod_names=[osd_pod.name for osd_pod in pod.get_osd_pods()], timeout=timeout
+    )
```
Comment (yitzhak12) [inline ocs_ci/ocs/node.py:1480]: I am not sure about this change... If the osd pods are running, are we sure that the new osd node should be running?

File: ocs_ci/ocs/machine.py [hunk 2]
```diff
@@ -157,27 +156,36 @@ def delete_machine_and_check_state_of_new_spinned_machine(machine_name):
...
-            .get("annotations")
-            .get("machine.openshift.io/instance-state")
+        new_machine_name = new_machine[0].name
+        log.info(f"Checking the state of new spinned machine {new_machine_name}")
+        new_machine[0].ocp.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            resource_name=new_machine_name,
+            column="PHASE",
+            timeout=600,
+            sleep=10,
```
Comment (yitzhak12) [inline ocs_ci/ocs/machine.py:183]: If we use `timeout=600`, maybe we can increase the `sleep` value to `20` or `30`?

Comment (sidhant-agrawal) [reply ocs_ci/ocs/machine.py]: done

**Comment by sidhant-agrawal** [comment]
> [Job UNSTABLE](https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-trigger-test-pr/1069//testReport) (some or all tests failed).

> [Job UNSTABLE](https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-trigger-test-pr/1071//testReport) (some or all tests failed).

The first PR validations failed due to #4551. The test is failing on AWS itself with the master branch code. It's not related to code change in this PR and hence raised a separate issue.

>
> [Job PASSED](https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-trigger-test-pr/1076//testReport).
 
> [Job PASSED](https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-trigger-test-pr/1080//testReport).

Other tests are passed on AWS IPI and vSphere IPI.
Marking PR as verified.

File: tests/manage/z_cluster/nodes/test_node_replacement_proactive.py [hunk 3]
```diff
@@ -88,32 +88,33 @@ def delete_and_create_osd_node(osd_node_name):
...
-            pytest.fail(msg_invalid)
-    elif config.ENV_DATA["platform"].lower() == constants.VSPHERE_PLATFORM:
+    if config.ENV_DATA["deployment_type"] == "ipi":
         if is_lso_cluster():
-            new_node_name = node.delete_and_create_osd_node_vsphere_upi_lso(
-                osd_node_name, use_existing_node=False
+            # TODO: Implement functionality for Internal-Attached devices mode
+            # once ocs-ci issue #4545 is resolved
+            # https://github.com/red-hat-storage/ocs-ci/issues/4545
+            raise NotImplementedError(
```
Comment (ebenahar) [inline tests/manage/z_cluster/nodes/test_node_replacement_proactive.py:97]: shouldn't we skip? throwing an exception means this test will fail

Comment (sidhant-agrawal) [reply tests/manage/z_cluster/nodes/test_node_replacement_proactive.py]: Right.
I've modified the code. Added skip instead of raising an exception.

**Review by ebenahar — DISMISSED** [review]

File: ocs_ci/ocs/node.py (hunk 1 above)
Comment (yitzhak12) [reply ocs_ci/ocs/node.py:1480]: Yes, I see. But maybe the time for the pods to reach the running state is less than the new osd node to appear in the new osd nodes. In this case, we shouldn't decrease the timeout in the next function from `timeout` to `60`.

Comment (sidhant-agrawal) [reply ocs_ci/ocs/node.py:1480]: ack, will increase the timeout in next function same as earlier, so as to not fail in the above scenario.

**Review by yitzhak12 — APPROVED** [review]
LGTM

**Review by ebenahar — APPROVED** [review]

**Pull request merged**